routers/user/updateStudentData.ts

```diff
@@ -0,0 +1,64 @@
+import { NextFunction, Request, Response } from "express";
+import { Roles } from "../../src/entities/types/Roles";
+import { ValidationError } from "../../utils/errorsHandler";
+import { myDataSource } from "../../config/database.configuration";
+import { StudentsDataInterface } from "../../src/entities/types/studentsData";
+import { UserPayloadData } from "../../utils/createTokens";
+import { StudentsData } from "../../src/entities/studentsData/studentsData.entity";
+import { validate } from "class-validator";
+
+type RequestAndPayloadUser = Request & UserPayloadData;
+
+const validationErrorOptions = {
+  validationError: {
+    target: false
+  }
+}
+
+export const updateStudentData = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
req should be RequestAndPayloadUser

Reply:
Jak opisywałem w innym pliku. Tam gdzie jesty wywoływana funkjca TS czepia się typów.